### PR TITLE
Add Json Adapted Commands

### PR DESCRIPTION
**Pull request by marcfyk**
There is a need to adapt `HistoryManager` for local storage. The plan is to have Jackson friendly commands and `HistoryManager` to be written to local storage in `JSON` format.

I have added `JsonAdaptedCommand` along with `AddressBook` related commands that are adapted for Jackson, along with `JsonSerializableHistoryManager`.

There are tests for the commands, however for `JsonSerializableHistoryManager`, tests currently only include the successful path, and has not tested any foreseeable errors that can come up.

### CODE REVIEW

File: src/main/java/seedu/jarvis/logic/commands/Command.java [hunk 1]
```diff
@@ -8,6 +8,13 @@
  */
 public abstract class Command {
 
+    /**
+     * Gets the command word of the command.
+     *
+     * @return {@code String} representation of the command word.
+     */
+    public abstract String getCommandWord();
```
Comment (ryanYtan) [inline src/main/java/seedu/jarvis/logic/commands/Command.java]: Alternative: can define `COMMAND_WORD` as a `protected final` field in `Command`, then have all sub-classes define their own `COMMAND_WORD` via the constructor, so you don't have to define `getCommandWord()` for every class.

Comment (marcfyk) [reply src/main/java/seedu/jarvis/logic/commands/Command.java]: Doesn't work for static fields, and the static field in each command class is useful for `JarvisParser` and testing.

File: src/main/java/seedu/jarvis/storage/history/JsonSerializableHistoryManager.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    private final List<JsonAdaptedCommand> inverselyExecutedCommands = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonSerializableHistoryManager} with the given executed and inversely executed commands.
+     *
+     * @param executedCommands {@code List} of executed Jackson-Friendly commands.
+     * @param inverselyExecutedCommands {@code List} of inversely executed Jackson-Friendly commands
+     */
+    @JsonCreator
+    public JsonSerializableHistoryManager(@JsonProperty("executedCommands") List<JsonAdaptedCommand> executedCommands,
```
Comment (mrnewguy) [inline src/main/java/seedu/jarvis/storage/history/JsonSerializableHistoryManager.java]: This implementation looks promising, perhaps we should discuss and do a minimal implementation of the storage by this weekend? My concern is that all the nesting in the json files might take some time to figure out.

Comment (marcfyk) [reply src/main/java/seedu/jarvis/storage/history/JsonSerializableHistoryManager.java]: Yes, the biggest hindrance is the nested objects that the commands have to store. Perhaps we can finalize certain commands so we can start working on `jackson` friendly classes for the other features.

**Pull request merged**